src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/PrereqsTests.cs

```diff
@@ -0,0 +1,249 @@
+                actualPackages.First(pkg => pkg.PackageName == expectedPackage.PackageName);
+
+            IEnumerable<string> expectedVersionDuplicates = expectedPackage.Versions
+                .GroupBy(ver => ver)
+                .Where(group => group.Count() > 1)
+                .Select(group => group.Key);
+
+            _outputHelper.WriteLine($"Verifying the versions for package '{expectedPackage.PackageName}' do not have duplicates. If this fails, the prereqs project at '{prereqsProjPath}' needs to be updated to remove duplicate versions for this package.");
+            Assert.Empty(expectedVersionDuplicates);
+
+            // During the development cycle, there can be incoherency leading to package versions which aren't consistent across packages.
+            // For that reason, we can only compare package versions whose values are statically defined in the prereqs project. Package
+            // versions which use a variable are considered to be dynamic and ignored.
+
+            IEnumerable<string> expectedDynamicVersions = expectedPackage.Versions
+                .Where(ver => _dynamicVersionRegex.IsMatch(ver))
+                .ToList();
+
+            Assert.True(expectedDynamicVersions.Count() <= 1,
+                $"Package '{expectedPackage.PackageName}' should not have more than 1 dynamic version in the prereqs project at '{prereqsProjPath}'. Dynamic versions defined: {string.Join(',', expectedDynamicVersions)}");
+
+            // Due to the potential for build incoherency, there can be multiple dynamic package versions in the output for a given package
+            // (the incoherency can lead to dependencies on different versions of the same package). But in the expected prereqs project,
+            // that package would only be listed once. So it's not possible to do a comparison of the number of versions between the expected
+            // and actual results when dynamic versions are present.
+            if (!expectedDynamicVersions.Any())
+            {
+                _outputHelper.WriteLine($"Verifying the actual number of versions associated with package '{expectedPackage.PackageName}' are the same as the expected. If this fails, the prereqs project at '{prereqsProjPath}' needs to be updated to reflect the actual version numbers being outputted by the smoke tests.");
+                Assert.Equal(expectedPackage.Versions.Count(), actualPackage.Versions.Count());
+            }
+
+            IEnumerable<string> expectedStaticVersions = expectedPackage.Versions
+                .Except(expectedDynamicVersions)
+                .OrderBy(ver => ver)
+                .ToList();
+
+            // Use the following heuristic for determining whether an outputted package's version represents a dynamic version:
+            //  - Package name begins with "Microsoft" or "System"
+            //      * Major/Minor version matches the version of the SDK
+            //      * Contains a preview version label
```

Comment:
This comment doesn't seem to align with the code.  There is the runtime and aspnet version case as well.

Reply:
Fixed in https://github.com/dotnet/installer/pull/16359/commits/e5733289e85309509128c2b4dfd12396ba599bdd